lib_eio/process.ml

```diff
@@ -141,7 +141,8 @@ let run t ?cwd ?stdin ?stdout ?stderr ?(is_success = Int.equal 0) ?env ?executab
     let ex = err (Child_error status) in
     raise (Exn.add_context ex "running command: %a" pp_args args)
 
-let pipe (type tag) ~sw ((Resource.T (v, ops)) : [> tag mgr_ty] r) =
+let pipe (type tag) ~sw ((Resource.T (v, ops)) : [> tag mgr_ty] r) :
+  ([< Flow.source_ty | Resource.close_ty ] r * [< Flow.sink_ty | Resource.close_ty ] r) =
```

Comment:
What is this change for? Did you mean to include it?

I suspect what you really want here is:
```ocaml
let pipe (type tag) ~sw ((Resource.T (v, ops)) : [> tag mgr_ty] r) =
  let module X = (val (Resource.get ops Pi.Mgr)) in
  let r, w = X.pipe v ~sw in
  let r = (r : [Flow.source_ty | Resource.close_ty] r :> [< Flow.source_ty | Resource.close_ty] r) in
  let w = (w : [Flow.sink_ty   | Resource.close_ty] r :> [< Flow.sink_ty   | Resource.close_ty] r) in
  r, w
```

Then most of the other changes can go away.

(it's annoying OCaml requires this to be written out in full; you'd think it could do the cast automatically)